frontend/src/components/pages/UserProfilePage.tsx

```diff
@@ -244,6 +295,22 @@ const UserProfilePage = () => {
           buttonMessage={MANAGE_USERS_TABLE_DELETE_USER_BUTTON}
         />
       )}
+      {addNewLanguage && (
+        <NewLanguageModal
+          addNewTest={addNewTest}
+          approvedLanguagesTranslation={approvedLanguagesTranslation}
+          isOpen={addNewLanguage}
+          onClose={closeNewLanguageModal}
+        />
+      )}
+      <ConfirmationModal
+        confirmation={alertNewTest}
+        onClose={closeNewTestModal}
+        onConfirmationClick={() => history.push("/")}
+        confirmationHeading="New Book Test Added"
+        confirmationMessage="The book test to add a new language has been added to the homepage. Please take one of the two tests available."
```

Comment:
Didn't use the existing new book test heading and message consts because they differ from the Figma.

Reply:
hmm the wording here is a bit confusing - it says "Please take one of the two tests available." but there's only one test created right

Reply:
rip, thats a mistake on designers + me. Can you use the existing new story translation test added copy from the level up flow?